heaps/heap_sort.py

```diff
@@ -1,8 +1,15 @@
-
+from heaps.min_heap import MinHeap
 
 def heap_sort(list):
     """ This method uses a heap to sort an array.
-        Time Complexity:  ?
-        Space Complexity: ?
+        Time Complexity:  O(nlogn)
+        Space Complexity: O(n)
     """
-    pass
\ No newline at end of file
+    heap = MinHeap()
+    sorted = []
+    for item in list:
+        heap.add(item)
+    for i in range(len(list)):
```

Comment:
In this situation, since we built the heap, we also "know" the number of items in the heap. So it works to iterate using knowledge about the list. But if we were pulling things out of a heap more generally, we would want to make use of the `empty` helper as follows:

```py
    sorted = []
    while not heap.empty():
        sorted.append(heap.remove())

    return sorted
```